src/core/simulation_results.jl

```diff
@@ -1,517 +1,378 @@
+        @info "Requested new times"
+        additional_values = _get_store_value(res, field, names, additional_timestamps)
+        for (var_name, df_dict) in additional_values
+            merge!(results_dict[var_name], df_dict)
         end
+    elseif isempty(additional_timestamps) && isempty(additional_names)
+    else
+        error()
     end
-    time_stamp[!, :Range] = convert(Array{Dates.DateTime}, time_stamp[!, :Range])
-    file_path = dirname(references[stage][variable[1]][1, :File_Path])
-    optimizer = read_json(joinpath(file_path, "optimizer_log.json"))
-    obj_value = Dict{Symbol, Any}(:OBJECTIVE_FUNCTION => optimizer["obj_value"])
-    duals = _read_references(duals, dual, stage, step, references, time_length)
-    param_values = _read_references(params, param, stage, step, references, time_length)
-    return SimulationResults(
-        base_power,
-        variables,
-        obj_value,
-        optimizer,
-        time_stamp,
-        duals,
-        results_folder,
-        param_values,
-    )
+    return results_dict
 end
 
 """
-    load_simulation_results(file_path, stage)
-
-This function goes through the reference table of file paths and
-aggregates the results over time into a struct of type OperationsProblemResults
-
-# Arguments
-- `file_path::String`: the file path to the dated folder with the raw results
-- `stage_number::String`: The stage of the results getting parsed
-
-# Example
-```julia
-execute!(simulation)
-results = load_simulation_results("file_path", "stage_name")
-```
-# Accepted Key Words
-"""
-function load_simulation_results(path::String, stage_name::String, kwargs...)
-    sim_results = deserialize_sim_output(path)
-    load_simulation_results(sim_results, stage_name; kwargs...)
-end
-"""
-    load_simulation_results(sim_output, stage)
-
-This function goes through the reference table of file paths and
-aggregates the results over time into a struct of type OperationsProblemResults
-
-# Arguments
-- `sim_output::SimulationResultsReference`: the container for the reference dictionary created in execute!
-- `stage_number::String`: The stage of the results getting parsed
+    Returns the values for the requested variable names. It keeps requests when performing multiple retrievals. Accepts a vector of names for the return of the values
 
-# Example
-```julia
-sim_output = execute!(simulation)
-results = load_simulation_results(sim_output, "stage_name")
-```
-# Accepted Key Words
+    # Accepted Key Words
+    - `initial_time::Dates.DateTime` : initial of the requested results
+    - `count::Int`: Number of results
 """
-function load_simulation_results(
-    sim_output::SimulationResultsReference,
-    stage_name::String;
-    kwargs...,
+function get_variables_values!(
+    res::SimulationResults,
+    names::Vector{Symbol};
+    initial_time::Union{Nothing, Dates.DateTime} = nothing,
+    count::Union{Int, Nothing} = nothing,
```

Comment:
Concerns:
1. It looks like the default behavior is to load all data into memory, which will grow unbounded. That does not seem ideal.
2. Suppose the user calls this function with the first initial_time and a count of 10. Then calls it again with 11 - 20. I don't see where you purge the old values. Will the data in memory grow unbounded? The docstrings don't address whether count applies to previously-read data.

The reads from the store should be relatively fast, especially if we can keep the .h5 file handle open. Would it be better to try that out first? This is a bit of complexity that may not be required which could also cause memory issues. A user doing plots may not notice the slightly slower behavior.